packages/react-google-maps-api/src/GoogleMap.tsx

```diff
@@ -65,101 +65,102 @@ const updaterMap = {
 }
 
 interface GoogleMapState {
-  map: google.maps.Map | null
+  map: google.maps.Map | null;
 }
 
 interface GoogleMapProps {
-  id: string
-  reuseSameInstance?: boolean
-  mapContainerStyle?: React.CSSProperties
-  mapContainerClassName?: string
-  options?: google.maps.MapOptions
-  extraMapTypes?: google.maps.MapType[]
-  center?: google.maps.LatLng | google.maps.LatLngLiteral
-  clickableIcons?: boolean
-  heading?: number
-  mapTypeId?: string
-  streetView?: google.maps.StreetViewPanorama
-  tilt?: number
-  zoom?: number
-  onClick?: (e: MouseEvent) => void
-  onDblClick?: (e: MouseEvent) => void
-  onDrag?: () => void
-  onDragEnd?: () => void
-  onDragStart?: () => void
-  onMapTypeIdChanged?: () => void
-  onMouseMove?: (e: MouseEvent) => void
-  onMouseOut?: (e: MouseEvent) => void
-  onMouseOver?: (e: MouseEvent) => void
-  onRightClick?: (e: MouseEvent) => void
-  onTilesLoaded?: () => void
-  onBoundsChanged?: () => void
-  onCenterChanged?: () => void
-  onHeadingChanged?: () => void
-  onIdle?: () => void
-  onProjectionChanged?: () => void
-  onResize?: () => void
-  onTiltChanged?: () => void
-  onZoomChanged?: () => void
-  onLoad: (map: google.maps.Map) => void | Promise<void>
+  id?: string;
+  reuseSameInstance?: boolean;
+  mapContainerStyle?: React.CSSProperties;
+  mapContainerClassName?: string;
+  options?: google.maps.MapOptions;
+  extraMapTypes?: google.maps.MapType[];
+  center?: google.maps.LatLng | google.maps.LatLngLiteral;
+  clickableIcons?: boolean;
+  heading?: number;
+  mapTypeId?: string;
+  streetView?: google.maps.StreetViewPanorama;
+  tilt?: number;
+  zoom?: number;
+  onClick?: (e: MouseEvent) => void;
+  onDblClick?: (e: MouseEvent) => void;
+  onDrag?: () => void;
+  onDragEnd?: () => void;
+  onDragStart?: () => void;
+  onMapTypeIdChanged?: () => void;
+  onMouseMove?: (e: MouseEvent) => void;
+  onMouseOut?: (e: MouseEvent) => void;
+  onMouseOver?: (e: MouseEvent) => void;
+  onRightClick?: (e: MouseEvent) => void;
+  onTilesLoaded?: () => void;
+  onBoundsChanged?: () => void;
+  onCenterChanged?: () => void;
+  onHeadingChanged?: () => void;
+  onIdle?: () => void;
+  onProjectionChanged?: () => void;
+  onResize?: () => void;
+  onTiltChanged?: () => void;
+  onZoomChanged?: () => void;
+  onLoad?: (map: google.maps.Map) => void | Promise<void>;
+  onUnmount?: (map: google.maps.Map) => void | Promise<void>;
 }
 
 export class GoogleMap extends React.PureComponent<
   GoogleMapProps,
   GoogleMapState
 > {
-  public static defaultProps: GoogleMapProps = {
```

Comment:
why remove default props?

Reply:
first of all, if prop is undefined, it is marked as `propName?: string; in interface declaration.

in this case we do not need default props, especially noop functions.

Reply:
I agree it's just a matter of taste weather to use defaultProps with noop or a condition checking if it is defined. But just FYI the way Typescript handles default props is that you remove the question mark and the prop is still optional because it has a default. I tested thia behavior and it works. But again, it works this way as well

Reply:
I prefer to be consistent) The only one issue which we got, where we could not get rid default props, is functional components, which requires some props default value to be an object or array. If you so default objects or array in props destructuring, each time those props has new object or array, which causing component to re-render. And defaultProps helps with avoiding this.